packages/async-flow/src/types.d.ts

```diff
@@ -46,11 +46,18 @@ export type GuestOf<F extends HostAsyncFuncWrapper> = F extends (
   ? (...args: A) => Promise<R>
   : F;
 
+// from https://github.com/sindresorhus/type-fest/blob/main/source/simplify.d.ts
+type Simplify<T> = { [KeyType in keyof T]: T[KeyType] } & {};
+
 /**
  * Convert an entire Guest interface into what the host will implement.
  */
 type HostInterface<T> = {
-  [K in keyof T]: HostOf<T[K]>;
+  [K in keyof T]: T[K] extends CallableFunction
+    ? HostOf<T[K]>
+    : T[K] extends Record<string, any>
+      ? Simplify<HostInterface<T[K]>>
+      : T[K];
```

Comment:
Only passable go through the membrane

```suggestion
      : T[K] extends Passable
        ? T[K]
        : never;
```